src/tribler/core/utilities/process_manager/manager.py

```diff
@@ -35,13 +35,31 @@ def get_global_process_manager() -> Optional[ProcessManager]:
 
 
 class ProcessManager:
-    def __init__(self, root_dir: Path, current_process: TriblerProcess, db_filename: str = DB_FILENAME):
+    def __init__(self, root_dir: Path, db_filename: str = DB_FILENAME):
         self.logger = logger  # Used by the `with_retry` decorator
         self.root_dir = root_dir
         self.db_filepath = root_dir / db_filename
         self.connection: Optional[sqlite3.Connection] = None
-        self.current_process = current_process
-        current_process.manager = self
+        self._current_process: Optional[TriblerProcess] = None
+
+    @with_retry
+    def setup_current_process(self, kind: ProcessKind, has_lock: bool, creator_pid: Optional[int] = None):
+        current_process = TriblerProcess.current_process(manager=self, kind=kind, has_lock=has_lock,
+                                                         creator_pid=creator_pid)
+        self._current_process = current_process
+        with self.connect():
+            if current_process.primary:
+                primary_process = self.get_primary_process(current_process.kind)
+                if primary_process is not None:
+                    raise RuntimeError(f'Previous primary process still active: {primary_process}. '
+                                       f'Current process: {current_process}')
```

Comment:
NIT: can be simplified

```suggestion
                if primary_process := self.get_primary_process(current_process.kind)
                    raise RuntimeError(f'Previous primary process still active: {primary_process}. '
                                       f'Current process: {current_process}')
```